When the modulus of a negative number is calculated, the result will either be negative or zero. Thus, comparing the modulus of a variable for equality with a positive number (or a negative one) could result in false negatives. 

<h2>Noncompliant Code Example</h2>
<pre>
public boolean isOdd(int x) {
  return x % 2 == 1;  // Noncompliant; if x is negative, x % 2 == -1
}
</pre>

<h2>Compliant Solution</h2>
<pre>
public boolean isOdd(int x) {
  return x %2 != 0;
}
</pre>

or

<pre>
public boolean isOdd(int x) {
  return Math.abs(x %2) != 1;
}
</pre>
